bigtable/client/internal/table.cc

```diff
@@ -114,6 +122,15 @@ std::vector<FailedMutation> Table::BulkApply(BulkMutation&& mut,
 }
 
 RowReader Table::ReadRows(RowSet row_set, Filter filter, bool raise_on_error) {
+  if (!app_profile_id_.empty()) {
```

Comment:
In general, if `app_profile_id` is empty the `set_app_profile_id(app_profile_id)` has the same effect as not setting it for proto3.  So we can avoid these additional `if()` statements, right?